lib/raml.js

```diff
@@ -278,7 +278,9 @@
             var item, kind;
             kind = _arg[0], item = _arg[1];
             if (item instanceof nodes.FileNode) {
-              return schemaNodes.push(item);
+              if (kind.value === 'schema' && definedSchemaNames.indexOf(item.value) === -1) {
```

Comment:
I considered this, but I did not, because I do not want to disturb the code too much for an upstream PR. If the maintainer likes this shortening, I figured he'll add it in the process of accepting it.